### PR TITLE
dir-assert

### LINKED ISSUE #14: find or build a tool to check that the filesystem BEFORE and AFTER you run some code is exactly the same. this is a prerequisite to all others.

### PR DESCRIPTION
**Pull request by ghost**
Installing this which [compares the content of two directiories or files](https://crates.io/crates/dir-assert)

### CODE REVIEW

**Review by amiller68 — DISMISSED** [review]
Looks good to me

File: dataprep-pipelines/Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ license.workspace = true
 repository.workspace = true
 
 [dependencies]
+dir-assert = "0.2.0"
```
Comment (laudiacay) [inline dataprep-pipelines/Cargo.toml]: use the workspace version

File: dataprep-pipelines/tests/integration.rs [hunk 2]
```diff
@@ -3,9 +3,17 @@ use tokio::fs;
 
 use dataprep_pipelines;
 
+
+#[test]
+fn test_macro() {
+    // macro to determine if two directories/files are equal
+    assert_paths!(input_dir, unpacked_dir);
```
Comment (laudiacay) [inline dataprep-pipelines/tests/integration.rs]: thanks for the example- can you rebase and edit my integration test to use this?

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -16,11 +16,14 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
+    - uses: actions-rs/toolchain@v1
```
Comment (laudiacay) [inline .github/workflows/rust.yml]: https://github.com/banyancomputer/ipfs-proof-buddy/blob/main/.github/workflows/ci.yml
https://github.com/banyancomputer/.github/blob/master/.github/workflows/rust-codecov.yml

Comment (laudiacay) [reply .github/workflows/rust.yml]: @amiller68 ^^

File: dataprep-pipelines/Cargo.toml (hunk 1 above)
Comment (laudiacay) [reply dataprep-pipelines/Cargo.toml]: @thea-exe

**Review by laudiacay — CHANGES_REQUESTED** [review]

**Comment by laudiacay** [comment]
Closes #14 

also did you know you can automatically make merging this PR close an issue by putting that text ^^

**Comment by laudiacay** [comment]
ugh i did it wrong...

File: dataprep-pipelines/Cargo.toml (hunk 1 above)
Comment (laudiacay) [reply dataprep-pipelines/Cargo.toml]: say dir-assert.workspace = true
and then leave the vresion specified in the top level workspace cargo.toml

File: .github/workflows/rust.yml (hunk 3 above)
Comment (laudiacay) [reply .github/workflows/rust.yml]: rebasing off master will resolve this

File: dataprep-pipelines/tests/integration.rs [hunk 4]
```diff
@@ -45,11 +41,13 @@ async fn it_works_for_one_file() {
...
-    println!("checking results!");
-    let final_out_contents = fs::read(final_out).await.unwrap();
-    println!(
-        "final_out_contents: {:?}",
-        String::from_utf8(final_out_contents.clone())
-    );
-    assert_eq!(final_out_contents, b"test");
+
+    // let final_out_contents = fs::read(final_out).await.unwrap();
+    // println!(
```
Comment (laudiacay) [inline dataprep-pipelines/tests/integration.rs]: let's go ahead and delete this now

File: .github/workflows/rust.yml [hunk 5]
```diff
@@ -13,6 +13,7 @@ env:
 
 # Queue up our Rust jobs (these all use +nightly):
 jobs:
+
```
Comment (laudiacay) [inline .github/workflows/rust.yml]: kill whitespace

File: Cargo.toml [hunk 6]
```diff
@@ -11,6 +11,7 @@ repository = "https://github.com/banyancomputer/dataprep"
 
 [workspace.dependencies]
 clap = { version = "4.0.32", features = ["derive"] }
+dir-assert = "0.2.0"
```
Comment (laudiacay) [inline Cargo.toml]: make this a workspace.dev-dependencies

File: dataprep-pipelines/Cargo.toml [hunk 7]
```diff
@@ -7,6 +7,7 @@ license.workspace = true
 repository.workspace = true
 
 [dependencies]
+dir-assert.workspace = true
```
Comment (laudiacay) [inline dataprep-pipelines/Cargo.toml]: make this a dev-dependency

File: dataprep-pipelines/src/pipeline/pack_pipeline.rs [hunk 8]
```diff
@@ -8,6 +8,7 @@ use std::path::PathBuf;
 use std::sync::Arc;
 use tokio::sync::RwLock;
 use tokio_stream::StreamExt;
+use dir_assert as _;
```
Comment (laudiacay) [inline dataprep-pipelines/src/pipeline/pack_pipeline.rs]: when you make this a dev dependency you won't need this import anymore

**Review by laudiacay — CHANGES_REQUESTED** [review]

**Pull request merged**